src/LayerEditor/ui/panels/regions_panel.py

```diff
@@ -54,16 +57,15 @@ def select_regions(self, region_dict):
         self._selected_regions.update(region_dict)
         self.region_changed.emit()
 
-    def select_layer(self, layer_id):
+    def select_layer(self, layer):
         """ Select actual layer tab."""
-        assert False, "Not refactored"
-        self.block.gui_selected_layers[self.current_topology_id] = layer_id
+        self.block.gui_selected_layer = layer
 
     def add_region(self, name, dim):
         """ Add new region according to the provided data from the current tab and select it."""
-        reg = self.regions_model.add_region(name, dim)
-        self.block.gui_selected_layer.gui_selected_region = reg
-        self.block.set_region_to_selected_shapes(reg)
+        with undo.group("Add new region"):
```

Comment:
Should rather be at the model layer.

Reply:
Similarly there should be the delete operation.

Reply:
Both are in `LEModel`. Done.